### Supermarket Sales Data for the 1st Quarter of 2019

In this jupyter notebook, we will try to determine the answers to the following questions:
- Q1: Which branch had the highest sales?
- Q2: Which product line had the most sales?
- Q3: Which month had the highest sales?
- Q4: What was the highest selling product line per gender?
- Q5: What is the highest rated product line?

Link to Kaggle data set: https://www.kaggle.com/datasets/aungpyaeap/supermarket-sales

We will also try to create a sales dashboard for the same data set using Tableau Public.

Link to dashboard: https://public.tableau.com/views/SalesDashboard_16874201280550/Dashboard1?:language=en-US&:display_count=n&:origin=viz_share_link

Import necessary libraries and dataset/s:

In [2]:
import pandas as pd
import numpy as np
from datetime import datetime as dt

In [3]:
# get data
sales = pd.read_csv('supermarket_sales.csv')
# view first few rows of dataset
sales.head()

Invoice ID Branch       City Customer type  Gender  \
0  750-67-8428      A     Yangon        Member  Female   
1  226-31-3081      C  Naypyitaw        Normal  Female   
2  631-41-3108      A     Yangon        Normal    Male   
3  123-19-1176      A     Yangon        Member    Male   
4  373-73-7910      A     Yangon        Normal    Male   

             Product line  Unit price  Quantity   Tax 5%     Total       Date  \
0       Health and beauty       74.69         7  26.1415  548.9715   1/5/2019   
1  Electronic accessories       15.28         5   3.8200   80.2200   3/8/2019   
2      Home and lifestyle       46.33         7  16.2155  340.5255   3/3/2019   
3       Health and beauty       58.22         8  23.2880  489.0480  1/27/2019   
4       Sports and travel       86.31         7  30.2085  634.3785   2/8/2019   

    Time      Payment    cogs  gross margin percentage  gross income  Rating  
0  13:08      Ewallet  522.83                 4.761905       26.1415     9.1  
1  10:29         Cash   76.40                 4.761905        3.8200     9.6  
2  13:23  Credit card  324.31                 4.761905       16.2155     7.4  
3  20:33      Ewallet  465.76                 4.761905       23.2880     8.4  
4  10:37      Ewallet  604.17                 4.761905       30.2085     5.3

Check number of rows and columns:

In [4]:
print('Shape of DF: ',sales.shape)
print('Column count: ', len(sales.columns))
print('Row count: ', len(sales.index))

Shape of DF:  (1000, 17)
Column count:  17
Row count:  1000


Check the list of 17 column names:

In [5]:
n = 1
for column in sales.columns:
    print(n, column)
    n = n + 1

1 Invoice ID
2 Branch
3 City
4 Customer type
5 Gender
6 Product line
7 Unit price
8 Quantity
9 Tax 5%
10 Total
11 Date
12 Time
13 Payment
14 cogs
15 gross margin percentage
16 gross income
17 Rating


Generate the statistical summary of the dataframe:

In [6]:
sales.describe()

Unit price     Quantity       Tax 5%        Total        cogs  \
count  1000.000000  1000.000000  1000.000000  1000.000000  1000.00000   
mean     55.672130     5.510000    15.379369   322.966749   307.58738   
std      26.494628     2.923431    11.708825   245.885335   234.17651   
min      10.080000     1.000000     0.508500    10.678500    10.17000   
25%      32.875000     3.000000     5.924875   124.422375   118.49750   
50%      55.230000     5.000000    12.088000   253.848000   241.76000   
75%      77.935000     8.000000    22.445250   471.350250   448.90500   
max      99.960000    10.000000    49.650000  1042.650000   993.00000   

       gross margin percentage  gross income      Rating  
count              1000.000000   1000.000000  1000.00000  
mean                  4.761905     15.379369     6.97270  
std                   0.000000     11.708825     1.71858  
min                   4.761905      0.508500     4.00000  
25%                   4.761905      5.924875     5.50000  
50%                   4.761905     12.088000     7.00000  
75%                   4.761905     22.445250     8.50000  
max                   4.761905     49.650000    10.00000

Check information about the dataframe:

In [7]:
sales.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1000 entries, 0 to 999
Data columns (total 17 columns):
 #   Column                   Non-Null Count  Dtype  
---  ------                   --------------  -----  
 0   Invoice ID               1000 non-null   object 
 1   Branch                   1000 non-null   object 
 2   City                     1000 non-null   object 
 3   Customer type            1000 non-null   object 
 4   Gender                   1000 non-null   object 
 5   Product line             1000 non-null   object 
 6   Unit price               1000 non-null   float64
 7   Quantity                 1000 non-null   int64  
 8   Tax 5%                   1000 non-null   float64
 9   Total                    1000 non-null   float64
 10  Date                     1000 non-null   object 
 11  Time                     1000 non-null   object 
 12  Payment                  1000 non-null   object 
 13  cogs                     1000 non-null   float64
 14  gross margin percentage  

Based on the data above, we need to convert some datatypes later on.

Check for null values:

In [8]:
sales.isnull().sum()

Invoice ID                 0
Branch                     0
City                       0
Customer type              0
Gender                     0
Product line               0
Unit price                 0
Quantity                   0
Tax 5%                     0
Total                      0
Date                       0
Time                       0
Payment                    0
cogs                       0
gross margin percentage    0
gross income               0
Rating                     0
dtype: int64

Based on the output above, our dataframe does not have null values.

We need to convert the Date and Time columns to a datetime format. I imported the datetime library above.

In [9]:
sales.dtypes

Invoice ID                  object
Branch                      object
City                        object
Customer type               object
Gender                      object
Product line                object
Unit price                 float64
Quantity                     int64
Tax 5%                     float64
Total                      float64
Date                        object
Time                        object
Payment                     object
cogs                       float64
gross margin percentage    float64
gross income               float64
Rating                     float64
dtype: object

In [10]:
sales['Date']

0       1/5/2019
1       3/8/2019
2       3/3/2019
3      1/27/2019
4       2/8/2019
         ...    
995    1/29/2019
996     3/2/2019
997     2/9/2019
998    2/22/2019
999    2/18/2019
Name: Date, Length: 1000, dtype: object

In [11]:
sales['Time']

0      13:08
1      10:29
2      13:23
3      20:33
4      10:37
       ...  
995    13:46
996    17:16
997    13:22
998    15:33
999    13:28
Name: Time, Length: 1000, dtype: object

As seen above, both Date and Time columns are dtype: objects. Let's convert them to datetime below:

In [12]:
sales['Date'] = pd.to_datetime(sales['Date'])
sales['Date']

0     2019-01-05
1     2019-03-08
2     2019-03-03
3     2019-01-27
4     2019-02-08
         ...    
995   2019-01-29
996   2019-03-02
997   2019-02-09
998   2019-02-22
999   2019-02-18
Name: Date, Length: 1000, dtype: datetime64[ns]

In [13]:
sales['Time'] = pd.to_datetime(sales['Time'])
sales['Time']

0     2023-07-01 13:08:00
1     2023-07-01 10:29:00
2     2023-07-01 13:23:00
3     2023-07-01 20:33:00
4     2023-07-01 10:37:00
              ...        
995   2023-07-01 13:46:00
996   2023-07-01 17:16:00
997   2023-07-01 13:22:00
998   2023-07-01 15:33:00
999   2023-07-01 13:28:00
Name: Time, Length: 1000, dtype: datetime64[ns]

In [14]:
sales.dtypes

Invoice ID                         object
Branch                             object
City                               object
Customer type                      object
Gender                             object
Product line                       object
Unit price                        float64
Quantity                            int64
Tax 5%                            float64
Total                             float64
Date                       datetime64[ns]
Time                       datetime64[ns]
Payment                            object
cogs                              float64
gross margin percentage           float64
gross income                      float64
Rating                            float64
dtype: object

Now, Date and Time columns have the appropriate dtype.

Check unique values per column:

In [15]:
sales.nunique()

Invoice ID                 1000
Branch                        3
City                          3
Customer type                 2
Gender                        2
Product line                  6
Unit price                  943
Quantity                     10
Tax 5%                      990
Total                       990
Date                         89
Time                        506
Payment                       3
cogs                        990
gross margin percentage       1
gross income                990
Rating                       61
dtype: int64

We have 1000 unique which means we have no duplicates. We are ready to proceed with analysis.

### Data Analysis
Visualization using Tableau Public:
https://public.tableau.com/views/SalesDashboard_16874201280550/Dashboard1?:language=en-US&:display_count=n&:origin=viz_share_link

##### Q1: Which branch had the highest sales?

In [39]:
# Again, let's check our data
sales.head()

Invoice ID Branch       City Customer type  Gender  \
0  750-67-8428      A     Yangon        Member  Female   
1  226-31-3081      C  Naypyitaw        Normal  Female   
2  631-41-3108      A     Yangon        Normal    Male   
3  123-19-1176      A     Yangon        Member    Male   
4  373-73-7910      A     Yangon        Normal    Male   

             Product line  Unit price  Quantity   Tax 5%     Total       Date  \
0       Health and beauty       74.69         7  26.1415  548.9715 2019-01-05   
1  Electronic accessories       15.28         5   3.8200   80.2200 2019-03-08   
2      Home and lifestyle       46.33         7  16.2155  340.5255 2019-03-03   
3       Health and beauty       58.22         8  23.2880  489.0480 2019-01-27   
4       Sports and travel       86.31         7  30.2085  634.3785 2019-02-08   

                 Time      Payment    cogs  gross margin percentage  \
0 2023-07-01 13:08:00      Ewallet  522.83                 4.761905   
1 2023-07-01 10:29:00         Cash   76.40                 4.761905   
2 2023-07-01 13:23:00  Credit card  324.31                 4.761905   
3 2023-07-01 20:33:00      Ewallet  465.76                 4.761905   
4 2023-07-01 10:37:00      Ewallet  604.17                 4.761905   

   gross income  Rating  Year     Month Month_Year  
0       26.1415     9.1  2019   January 2019-01-01  
1        3.8200     9.6  2019     March 2019-03-01  
2       16.2155     7.4  2019     March 2019-03-01  
3       23.2880     8.4  2019   January 2019-01-01  
4       30.2085     5.3  2019  February 2019-02-01

In [38]:
# Check the branches available in our data set
print('Branches: ',sales['Branch'].unique().tolist())

Branches:  ['A', 'C', 'B']


In [18]:
# Get total sales per branch
salesByCity = sales.groupby(['Branch'])['Total'].sum()
salesByCity

Branch
A    106200.3705
B    106197.6720
C    110568.7065
Name: Total, dtype: float64

Answer: Branch C had the most sales compared to all branches. Total Sales for Branch C was $110568.71.

Check visualization (Total Sales by Branch): https://public.tableau.com/views/SalesDashboard_16874201280550/Dashboard1?:language=en-US&:display_count=n&:origin=viz_share_link

##### Q2: Which product line had the most sales?

In [21]:
# Let's check all the product lines we have in our dataset
print('Product Lines: ',sales['Product line'].unique())

Product Lines:  ['Health and beauty' 'Electronic accessories' 'Home and lifestyle'
 'Sports and travel' 'Food and beverages' 'Fashion accessories']


In [46]:
# Check the total sales grouped by Product line
productSales = sales.groupby(['Product line'])['Total'].sum()
productSales

Product line
Electronic accessories    54337.5315
Fashion accessories       54305.8950
Food and beverages        56144.8440
Health and beauty         49193.7390
Home and lifestyle        53861.9130
Sports and travel         55122.8265
Name: Total, dtype: float64

Answer: Food and beverages produced the most sales at $56,155.84

Check visualization (Sales by Product Line and Gender): https://public.tableau.com/views/SalesDashboard_16874201280550/Dashboard1?:language=en-US&:display_count=n&:origin=viz_share_link

##### Q3: Which month had the highest sales?

In [24]:
# Let's add 3 more columns, Month Name, Year, and Month Year so we can group by month.
sales['Year'] = sales['Date'].dt.year
sales['Month'] = sales['Date'].dt.month_name()
sales['Month_Year'] = pd.to_datetime(sales['Year'].astype(str) + sales['Month'], format='%Y%B')
sales.head()

Invoice ID Branch       City Customer type  Gender  \
0  750-67-8428      A     Yangon        Member  Female   
1  226-31-3081      C  Naypyitaw        Normal  Female   
2  631-41-3108      A     Yangon        Normal    Male   
3  123-19-1176      A     Yangon        Member    Male   
4  373-73-7910      A     Yangon        Normal    Male   

             Product line  Unit price  Quantity   Tax 5%     Total       Date  \
0       Health and beauty       74.69         7  26.1415  548.9715 2019-01-05   
1  Electronic accessories       15.28         5   3.8200   80.2200 2019-03-08   
2      Home and lifestyle       46.33         7  16.2155  340.5255 2019-03-03   
3       Health and beauty       58.22         8  23.2880  489.0480 2019-01-27   
4       Sports and travel       86.31         7  30.2085  634.3785 2019-02-08   

                 Time      Payment    cogs  gross margin percentage  \
0 2023-07-01 13:08:00      Ewallet  522.83                 4.761905   
1 2023-07-01 10:29:00         Cash   76.40                 4.761905   
2 2023-07-01 13:23:00  Credit card  324.31                 4.761905   
3 2023-07-01 20:33:00      Ewallet  465.76                 4.761905   
4 2023-07-01 10:37:00      Ewallet  604.17                 4.761905   

   gross income  Rating  Year     Month Month_Year  
0       26.1415     9.1  2019   January 2019-01-01  
1        3.8200     9.6  2019     March 2019-03-01  
2       16.2155     7.4  2019     March 2019-03-01  
3       23.2880     8.4  2019   January 2019-01-01  
4       30.2085     5.3  2019  February 2019-02-01

In [25]:
monthlySales = sales.groupby(['Month_Year'])['Total'].sum()
monthlySales

Month_Year
2019-01-01    116291.868
2019-02-01     97219.374
2019-03-01    109455.507
Name: Total, dtype: float64

Answer: Highest sales for 1st QTR was on January 2019.

Check visualization (Monthly Sales by Branch): https://public.tableau.com/views/SalesDashboard_16874201280550/Dashboard1?:language=en-US&:display_count=n&:origin=viz_share_link

##### Q4: What was the highest selling product line per gender?

In [35]:
productSoldMF = sales.groupby(['Gender','Product line'])['Total'].sum()
productSoldMF

Gender  Product line          
Female  Electronic accessories    27102.0225
        Fashion accessories       30437.4000
        Food and beverages        33170.9175
        Health and beauty         18560.9865
        Home and lifestyle        30036.8775
        Sports and travel         28574.7210
Male    Electronic accessories    27235.5090
        Fashion accessories       23868.4950
        Food and beverages        22973.9265
        Health and beauty         30632.7525
        Home and lifestyle        23825.0355
        Sports and travel         26548.1055
Name: Total, dtype: float64

Answer: For Female, Food and Beverages had the highest sales with $33,170.92 and for Male, Health and beauty had the highest sales with $30,632.75

Check visualization (Sales by Product Line and Gender): https://public.tableau.com/views/SalesDashboard_16874201280550/Dashboard1?:language=en-US&:display_count=n&:origin=viz_share_link

##### Q5: What is the highest rated product line?

In [37]:
branchRating = sales.groupby(['Product line'])['Rating'].mean()
branchRating

Product line
Electronic accessories    6.924706
Fashion accessories       7.029213
Food and beverages        7.113218
Health and beauty         7.003289
Home and lifestyle        6.837500
Sports and travel         6.916265
Name: Rating, dtype: float64

Answer: Food and beverages has the highest average rating with a score of 7.11, followed by Fashion accessories at 7.03.

Check visualization (Average Rating of Product Lines): https://public.tableau.com/views/SalesDashboard_16874201280550/Dashboard1?:language=en-US&:display_count=n&:origin=viz_share_link